ITISFoundation/osparc-simcore
# ✨ Auto inject osparc environments to dynamic services

## What do these changes do?

Injects automatically 
```yaml
environment:
    OSPARC_API_BASE_URL: "$OSPARC_VARIABLE_API_HOST"
    OSPARC_API_KEY: "$OSPARC_VARIABLE_API_KEY"
    OSPARC_API_SECRET: "$OSPARC_VARIABLE_API_SECRET"
    OSPARC_NODE_ID: "$OSPARC_VARIABLE_NODE_ID"
    OSPARC_STUDY_ID: "$OSPARC_VARIABLE_STUDY_UUID"
```
to **all** services in a dynamic-service compose. I.e. all dynamic services get automatically access to these environs. NOTE that the key/secrets are customised per user.


@sanderegg @mguidon should we only inject in the "main" service, i.e. the one specified by `simcore.service.container-http-entrypoint`?

## Related issue/s

- resolves #5925 

## How to test

- driving tests `services/director-v2/tests/unit/test_modules_osparc_variables.py`
- Manual testing:
   - Open jupyter (or any other dynamic service).
   - `printenv | sort | grep OSPARC_` and check that all the env-vars above are defined
![Uploading image.png…]()


## Dev-ops checklist
